pkg/relay/ethereum/tx.go

```diff
@@ -278,3 +268,69 @@ func (c *Chain) TxAcknowledgement(opts *bind.TransactOpts, msg *chantypes.MsgAck
 		ProofHeight:     pbToHandlerHeight(msg.ProofHeight),
 	})
 }
+
+func (c *Chain) SendTx(opts *bind.TransactOpts, msg sdk.Msg, noSend bool) (*gethtypes.Transaction, error) {
+	logger := c.GetChainLogger()
+	opts.NoSend = noSend
+	var (
+		tx  *gethtypes.Transaction
+		err error
+	)
+	switch msg := msg.(type) {
+	case *clienttypes.MsgCreateClient:
+		tx, err = c.TxCreateClient(opts, msg)
+	case *clienttypes.MsgUpdateClient:
+		tx, err = c.TxUpdateClient(opts, msg)
+	case *conntypes.MsgConnectionOpenInit:
+		tx, err = c.TxConnectionOpenInit(opts, msg)
+	case *conntypes.MsgConnectionOpenTry:
+		tx, err = c.TxConnectionOpenTry(opts, msg)
+	case *conntypes.MsgConnectionOpenAck:
+		tx, err = c.TxConnectionOpenAck(opts, msg)
+	case *conntypes.MsgConnectionOpenConfirm:
+		tx, err = c.TxConnectionOpenConfirm(opts, msg)
+	case *chantypes.MsgChannelOpenInit:
+		tx, err = c.TxChannelOpenInit(opts, msg)
+	case *chantypes.MsgChannelOpenTry:
+		tx, err = c.TxChannelOpenTry(opts, msg)
+	case *chantypes.MsgChannelOpenAck:
+		tx, err = c.TxChannelOpenAck(opts, msg)
+	case *chantypes.MsgChannelOpenConfirm:
+		tx, err = c.TxChannelOpenConfirm(opts, msg)
+	case *chantypes.MsgRecvPacket:
+		tx, err = c.TxRecvPacket(opts, msg)
+	case *chantypes.MsgAcknowledgement:
+		tx, err = c.TxAcknowledgement(opts, msg)
+	// case *transfertypes.MsgTransfer:
+	// 	err = c.client.transfer(msg)
+	default:
+		logger.Error("failed to send msg", errors.New("illegal msg type"), "msg", msg)
+		panic("illegal msg type")
+	}
+	return tx, err
+}
+
+func (c *Chain) EstimateGas(ctx context.Context, tx *gethtypes.Transaction) (uint64, error) {
+	from, err := gethtypes.Sender(gethtypes.NewEIP155Signer(tx.ChainId()), tx)
+	if err != nil {
+		return 0, err
+	}
+	to := tx.To()
+	value := tx.Value()
+	gas := tx.Gas()
+	gasPrice := tx.GasPrice()
+	data := tx.Data()
+	callMsg := ethereum.CallMsg{
+		From:     from,
+		To:       to,
+		Gas:      gas,
```

Comment:
You should not set `Gas` when invoking `eth_estimateGas`, right?
Please remove this and add `GasTipCap` and `GasFeeCap`.
ref: https://github.com/ethereum/go-ethereum/blob/e501b3b05db8e169f67dc78b7b59bc352b3c638d/accounts/abi/bind/base.go#L348-L356

Reply:
Remove gas, Add GasTipCap, GasFeeCap